src/Spout/Writer/WriterAbstract.php

```diff
@@ -344,20 +344,25 @@ public function addRowsWithStyle(array $dataRows, $style)
             throw new InvalidArgumentException('The "$style" argument must be a Style instance and cannot be NULL.');
         }
 
-        $this->addRows(array_map(function ($row) use ($style) {
+        foreach($dataRows as $row) {
+
             if (is_array($row)) {
-                return $this->createRowFromArray($row, $style);
+                $row = $this->createRowFromArray($row, $style);
             } elseif ($row instanceof Row) {
-                return $row;
+                $row->setStyle($style);
             } else {
                 throw new InvalidArgumentException();
             }
-        }, $dataRows));
+
+            $this->addRow($row);
+        }
 
         return $this;
     }
 
     /**
+     * @TODO: Move this into styleMerger
```

Comment:
I meant only move the `if (null === $defaultRowStyle)` part. The `merge` function should handle null param